### PR TITLE
Track the trihedron

### PR DESCRIPTION
**Pull request by eggrobin**
A first cut of #876. From the features side of things, the "show on navball" functionality needs to work differently in that case.
Subsequently, the implementation should be changed to properly integrate the velocity-dependent acceleration.

### CODE REVIEW

File: ksp_plugin_adapter/gl_lines.cs [hunk 1]
```diff
@@ -142,6 +142,7 @@ public enum Style {
     UnityEngine.Vector3 camera_position_in_world =
         ScaledSpace.ScaledToLocalSpace(camera.transform.position);
 
+
```
Comment (pleroy) [inline ksp_plugin_adapter/gl_lines.cs]: Remove extra empty line.

File: tools/journal_proto_processor.cpp [hunk 2]
```diff
@@ -425,6 +425,8 @@ void JournalProtoProcessor::ProcessRequiredMessageField(
 void JournalProtoProcessor::ProcessRequiredBoolField(
     FieldDescriptor const* descriptor) {
   field_cs_type_[descriptor] = "bool";
+  field_cs_marshal_[descriptor] = "MarshalAs(UnmanagedType.I1)";
+  field_cs_private_type_[descriptor] = "Byte";
```
Comment (pleroy) [inline tools/journal_proto_processor.cpp]: Please sync, this should not be in this PR.

File: ksp_plugin/flight_plan.cpp [hunk 3]
```diff
@@ -305,15 +305,51 @@ void FlightPlan::BurnLastSegment(NavigationManœuvre const& manœuvre) {
...
+            ephemeris_->FlowWithAdaptiveStep(segments_.back(),
+                                             movement.IntrinsicAcceleration(),
+                                             movement.final_time(),
+                                             adaptive_step_parameters_,
+                                             max_ephemeris_steps_per_frame,
+                                             /*last_point_only=*/false);
+        if (!reached_desired_final_time) {
+          anomalous_segments_ = 1;
+          return;
+        }
```
Comment (pleroy) [inline ksp_plugin/flight_plan.cpp]: Please add a test for the new code path.

File: ksp_plugin_test/manœuvre_test.cpp [hunk 4]
```diff
@@ -96,7 +96,8 @@ TEST_F(ManœuvreTest, TimedBurn) {
       /*initial_mass=*/2 * Kilogram,
       /*specific_impulse=*/1 * Newton * Second / Kilogram,
       /*direction=*/2 * e_y,
-      MakeMockDynamicFrame());
+      MakeMockDynamicFrame(),
+      /*is_inertially_fixed=*/true);
```
Comment (pleroy) [inline ksp_plugin_test/manœuvre_test.cpp]: Please add a test where this is false.

File: ksp_plugin/flight_plan.cpp [hunk 5]
```diff
@@ -305,15 +305,51 @@ void FlightPlan::BurnLastSegment(NavigationManœuvre const& manœuvre) {
...
+                                           adaptive_step_parameters_,
+                                           max_ephemeris_steps_per_frame,
+                                           /*last_point_only=*/false);
+      if (!reached_desired_final_time) {
+        anomalous_segments_ = 1;
+      }
+    } else {
+      // We decompose the manœuvre in smaller unguided manœuvres (movements),
+      // which are unguided.
+      // TODO(egg): eventually we should just compute the direction in the
```
Comment (eggrobin) [inline ksp_plugin/flight_plan.cpp]: #1485.

File: ksp_plugin/manœuvre_body.hpp [hunk 6]
```diff
@@ -173,6 +175,11 @@ OrthogonalMap<Frenet<Frame>, InertialFrame>
 }
 
 template<typename InertialFrame, typename Frame>
+inline bool Manœuvre<InertialFrame, Frame>::is_inertially_fixed() const {
```
Comment (eggrobin) [inline ksp_plugin/manœuvre_body.hpp]: Done.

File: ksp_plugin_adapter/gl_lines.cs (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin_adapter/gl_lines.cs]: Done.

File: ksp_plugin/flight_plan.cpp (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin/flight_plan.cpp]: Done.

**Pull request merged**